java/core/src/main/java/co/worklytics/psoxy/gateway/impl/ApiDataRequestHandler.java

```diff
@@ -982,5 +989,44 @@ JsonNode applyStringTransformToTree(JsonNode jsonNode,
         }
     }
 
+    /**
+     * Check if an exception or any of its causes is a SocketTimeoutException
+     */
+    private boolean isSocketTimeoutException(Throwable throwable) {
+        Throwable cause = throwable;
+        while (cause != null) {
+            if (cause instanceof java.net.SocketTimeoutException) {
+                return true;
+            }
+            cause = cause.getCause();
+        }
+        return false;
+    }
+
+    /**
+     * Build error response for network timeout issues
+     * Cannot distinguish between:
+     * 1) Network egress blocked from proxy (VPC/serverless connector misconfiguration)
+     * 2) Target API being unreachable/slow (their infrastructure issue)
+     */
+    private HttpEventResponse buildNetworkTimeoutErrorResponse(
```

Comment:
Just for personal preference; as this method is without a state and just a builder, I'd put it static ### 
```suggestion
    private static HttpEventResponse buildNetworkTimeoutErrorResponse(
```